fabric-block-api-v1/src/main/java/net/fabricmc/fabric/api/block/v1/FabricBlock.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.block.v1;
+
+import org.jetbrains.annotations.Nullable;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.Direction;
+import net.minecraft.world.BlockRenderView;
+import net.minecraft.world.World;
+
+/**
+ * General-purpose Fabric-provided extensions for {@link Block} subclasses.
+ *
+ * <p>Note: This interface is automatically implemented on all blocks via Mixin and interface injection.
+ *
+ * <p>Note to maintainers: Functions should only be added to this interface if they are general-purpose enough,
+ * to be evaluated on a case-by-case basis. Otherwise, they are better suited for more specialized APIs.
+ */
+public interface FabricBlock {
+	/**
+	 * Return the current appearance of the block, i.e. which block state this block reports to look like on a given side.
+	 *
+	 * <p>Common implementors are covers and facades, or any other mimic blocks that proxy another block's model.
+	 * These will want to override this method. In that case, make sure to carefully read the implementation guidelines below.
+	 *
+	 * <p>Common consumers are models with connected textures that wish to seamlessly connect to mimic blocks.
+	 * These will want to check the apparent block state using {@link FabricBlockState#getAppearance}.
+	 *
+	 * <p>Generally, the appearance will be queried from a nearby block,
+	 * identified by the optional {@code sourcePos} and {@code sourceState} parameters.
```

Comment:
Hmm, what's this needed for?

Reply:
This can be used by mods like chisel&bits when there's more than one block state per face.

Note: The `sourcePos` param was already in a similar CTM API upon request by the author of little tiles, and after discussion with Pepper we decided to also add the block state parameter.

Reply:
> This can be used by mods like chisel&bits when there's more than one block state per face.

Like "if queried by `sourceState` and I contain `sourceState`, let's report us appearing as that state" type of thing? Neat

Reply:
Yep, that's the idea !